In [1]:
import pandas as pd
import numpy as np
nsmes_df = pd.read_csv('NSMES1988.csv')
nsmes_df

Unnamed: 0  visits  nvisits  ovisits  novisits  emergency  hospital  \
0              1       5        0        0         0          0         1   
1              2       1        0        2         0          2         0   
2              3      13        0        0         0          3         3   
3              4      16        0        5         0          1         1   
4              5       3        0        0         0          0         0   
...          ...     ...      ...      ...       ...        ...       ...   
4401        4402      11        0        0         0          0         0   
4402        4403      12        0        0         0          0         0   
4403        4404      10        0       20         0          1         1   
4404        4405      16        1        0         0          0         0   
4405        4406       0        0        0         0          0         0   

         health  chronic      adl region  age  gender married  school  \
0       average        2   normal  other  6.9    male     yes       6   
1       average        2   normal  other  7.4  female     yes      10   
2          poor        4  limited  other  6.6  female      no      10   
3          poor        2  limited  other  7.6    male     yes       3   
4       average        2  limited  other  7.9  female     yes       6   
...         ...      ...      ...    ...  ...     ...     ...     ...   
4401    average        0   normal  other  8.4  female     yes       8   
4402    average        2   normal  other  7.8  female      no      11   
4403    average        5   normal  other  7.3    male     yes      12   
4404    average        0   normal  other  6.6  female     yes      12   
4405  excellent        0   normal  other  7.1    male     yes       0   

        income employed insurance medicaid  
0     2.881000      yes       yes       no  
1     2.747800       no       yes       no  
2     0.653200       no        no      yes  
3     0.658800       no       yes       no  
4     0.658800       no       yes       no  
...        ...      ...       ...      ...  
4401  2.249700       no       yes       no  
4402  5.813200       no       yes       no  
4403  3.877916       no       yes       no  
4404  3.877916       no       yes       no  
4405  6.596800      yes        no       no  

[4406 rows x 19 columns]

In [2]:
#checking the structural information of the data
#nsmes_df.info()
#nsmes_df.drop('Unnamed: 0', axis=1, inplace=True)# removed 'Unnamed: 0' which is not significant
nsmes_df
#Finding out if there are null values and non null values count
nsmes_df

null_counts = nsmes_df.isnull().sum()
print(null_counts.dtype) #There are no null values
print(null_counts)

int64
Unnamed: 0    0
visits        0
nvisits       0
ovisits       0
novisits      0
emergency     0
hospital      0
health        0
chronic       0
adl           0
region        0
age           0
gender        0
married       0
school        0
income        0
employed      0
insurance     0
medicaid      0
dtype: int64


In [3]:
#adding columns - salary*10000 and age*10

nsmes_df['act_age'] = nsmes_df['age']*10
nsmes_df['act_income'] = nsmes_df['income']*10000
nsmes_df

#nsmes_df['visits'].max()
#nsmes_df['ovisits'].max()
#nsmes_df['novisits'].max()
#nsmes_df['emergency'].max()
#nsmes_df['hospital'].max()
#nsmes_df['nvisits'].max()

#All these values are smaller hence we can save memory if we use unit8 instead of default int64 datatype



Unnamed: 0  visits  nvisits  ovisits  novisits  emergency  hospital  \
0              1       5        0        0         0          0         1   
1              2       1        0        2         0          2         0   
2              3      13        0        0         0          3         3   
3              4      16        0        5         0          1         1   
4              5       3        0        0         0          0         0   
...          ...     ...      ...      ...       ...        ...       ...   
4401        4402      11        0        0         0          0         0   
4402        4403      12        0        0         0          0         0   
4403        4404      10        0       20         0          1         1   
4404        4405      16        1        0         0          0         0   
4405        4406       0        0        0         0          0         0   

         health  chronic      adl  ...  age  gender married school    income  \
0       average        2   normal  ...  6.9    male     yes      6  2.881000   
1       average        2   normal  ...  7.4  female     yes     10  2.747800   
2          poor        4  limited  ...  6.6  female      no     10  0.653200   
3          poor        2  limited  ...  7.6    male     yes      3  0.658800   
4       average        2  limited  ...  7.9  female     yes      6  0.658800   
...         ...      ...      ...  ...  ...     ...     ...    ...       ...   
4401    average        0   normal  ...  8.4  female     yes      8  2.249700   
4402    average        2   normal  ...  7.8  female      no     11  5.813200   
4403    average        5   normal  ...  7.3    male     yes     12  3.877916   
4404    average        0   normal  ...  6.6  female     yes     12  3.877916   
4405  excellent        0   normal  ...  7.1    male     yes      0  6.596800   

      employed insurance medicaid act_age  act_income  
0          yes       yes       no    69.0    28810.00  
1           no       yes       no    74.0    27478.00  
2           no        no      yes    66.0     6532.00  
3           no       yes       no    76.0     6588.00  
4           no       yes       no    79.0     6588.00  
...        ...       ...      ...     ...         ...  
4401        no       yes       no    84.0    22497.00  
4402        no       yes       no    78.0    58132.00  
4403        no       yes       no    73.0    38779.16  
4404        no       yes       no    66.0    38779.16  
4405       yes        no       no    71.0    65968.00  

[4406 rows x 21 columns]

In [4]:
#Checking if there are any duplicate rows

duplicate_rows = nsmes_df.duplicated().value_counts()
duplicate_rows

#There are no duplicate rows present


False    4406
Name: count, dtype: int64

In [5]:
#Trying to analyze categorical values
nsmes_df['health'].unique()
nsmes_df['health'].value_counts()

nsmes_df['adl'].unique()
nsmes_df['adl'].value_counts()

nsmes_df['gender'].unique()
nsmes_df['gender'].value_counts()

nsmes_df['employed'].unique()
nsmes_df['employed'].value_counts()

nsmes_df['medicaid'].unique()
nsmes_df['medicaid'].value_counts()

nsmes_df['insurance'].unique()
nsmes_df['insurance'].value_counts()


insurance
yes    3421
no      985
Name: count, dtype: int64

In [6]:
#Exporting data to new csv file
nsmes_df.to_csv('NSMES1988_new.csv', index=False)

# **Visual Observations of the data**

**About the data set:** This is Patient's dataset about their visits to pysicians and hospital and information about their insurance and other paramters

**Size of the data:** 4406 * 19 columns

**Observations:**

1.   One of the columns named "Unnamed: 0" which was kind
of index and there was no other significance of it hence I dropped it.
2.   There are columns which count the patient's visits to hospital or pysician. I checked max values and found that they are very small and we do not need to use default data type "int64". Instead we can use unit8.


1.   There are no null values in data set. All the rows have all the data elements populated.
2.   Added 2 more columns as act_salary and act_age as they were divided by 10000 and 10 respectivly.


1.   Checked all the categorical values(all the columns in the dataset and found that they are clean in consitent)




